Emby.Server.Implementations/Data/SqliteItemRepository.cs

```diff
@@ -6133,5 +6147,167 @@ private MediaStream GetMediaStream(IReadOnlyList<IResultSetValue> reader)
 
             return item;
         }
+
+        public List<MediaAttachment> GetMediaAttachments(MediaAttachmentQuery query)
+        {
+            CheckDisposed();
+
+            if (query == null)
+            {
+                throw new ArgumentNullException(nameof(query));
+            }
+
+            var cmdText = "select "
+                        + string.Join(",", _mediaAttachmentSaveColumns)
+                        + " from mediaattachments where"
+                        + " ItemId=@ItemId";
+
+            if (query.Index.HasValue)
+            {
+                cmdText += " AND AttachmentIndex=@AttachmentIndex";
+            }
+
+            cmdText += " order by AttachmentIndex ASC";
+
+            var list = new List<MediaAttachment>();
+            using var connection = GetConnection(true);
+            using (var statement = PrepareStatement(connection, cmdText))
+            {
+                statement.TryBind("@ItemId", query.ItemId.ToByteArray());
+
+                if (query.Index.HasValue)
+                {
+                    statement.TryBind("@AttachmentIndex", query.Index.Value);
+                }
+
+                foreach (var row in statement.ExecuteQuery()) {
+                    list.Add(GetMediaAttachment(row));
+                }
+            }
+
+            return list;
+        }
+
+        public void SaveMediaAttachments(Guid id, List<MediaAttachment> attachments, CancellationToken cancellationToken)
+        {
+            CheckDisposed();
+            if (id == Guid.Empty)
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
+
+            if (attachments == null)
+            {
+                throw new ArgumentNullException(nameof(attachments));
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            using var connection = GetConnection();
+            connection.RunInTransaction(db =>
+            {
+                var itemIdBlob = id.ToByteArray();
+
+                db.Execute("delete from mediaattachments where ItemId=@ItemId", itemIdBlob);
+
+                InsertMediaAttachments(itemIdBlob, attachments, db);
+
+            }, TransactionMode);
+        }
+
+        private void InsertMediaAttachments(byte[] idBlob, List<MediaAttachment> attachments, IDatabaseConnection db)
+        {
+            var startIndex = 0;
+            var insertAtOnce = 10;
+
+            while (startIndex < attachments.Count)
+            {
+                var insertText = new StringBuilder(string.Format("insert into mediaattachments ({0}) values ", string.Join(",", _mediaAttachmentSaveColumns)));
+
+                var endIndex = Math.Min(attachments.Count, startIndex + insertAtOnce);
+
+                for (var i = startIndex; i < endIndex; i++)
+                {
+                    if (i != startIndex)
```

Comment:
Do as below, add it at the end of each iteration and reduce the length of the `StringBuilder` by 1 after the loop.